pkg/sonar/sonar.go

```diff
@@ -0,0 +1,31 @@
+package sonar
+
+import (
+	"path/filepath"
+
+	"github.com/magiconair/properties"
+)
+
+// TaskReportData ...
+// https://pkg.go.dev/github.com/magiconair/properties@v1.8.0?tab=doc#Properties.Decode
+type TaskReportData struct {
+	ProjectKey    string `properties:"projectKey"`
+	TaskID        string `properties:"ceTaskId"`
+	DashboardURL  string `properties:"dashboardUrl"`
+	TaskURL       string `properties:"ceTaskUrl"`
+	ServerURL     string `properties:"serverUrl"`
+	ServerVersion string `properties:"serverVersion"`
+}
+
+//ReadTaskReport ...
+func ReadTaskReport(workspace string) (result TaskReportData, err error) {
+	reportFile := filepath.Join(workspace, ".scannerwork", "report-task.txt")
+	// read file content
+	reportContent, err := properties.LoadFile(reportFile, properties.UTF8)
+	if err != nil {
+		return
+	}
+	// read content into struct
+	err = reportContent.Decode(&result)
```

Comment:
I would be in favor to check the error here, and if !nil, wrap in another error that is more descriptive to the user. I see from your unit test, that this could be "missing required key projectKey" for example, which seems to lack any context.

... unless of course the error is wrapped farther up the call chain.